### PR TITLE
Confirmation Modals improvements; closes #2613.

### LINKED ISSUE #2613: Add Confirmation Modals to Event Handlers for Destructive Processes
Currently a user can click on buttons (or perform keypresses) for destructive events, like deleting an entity or closing a form, and there is nothing to prevent that from happening if the click or keypress was accidental.

Therefore we need to add a confirmation step prior to performing any kind of destructive logic

- delete buttons
- modal cancel button
- modal close button
- ESC key (onKeyPress handler)

The [AntD Popconfirm component](https://ant.design/components/popconfirm/) is designed specifically for this use case

### PR DESCRIPTION
**Pull request by nicolad**
## Problem this Pull Request solves
- Closes: https://github.com/eventespresso/event-espresso-core/issues/2613

### CODE REVIEW

**Comment by nicolad** [comment]
Peek 2020-05-06 14-24

File: assets/src/application/ui/layout/modal/ModalWithAlert.tsx [hunk 1]
```diff
@@ -19,8 +19,10 @@ const ModalWithAlert: React.FC<Props> = ({ children, showAlertOnEscape, ...props
 	const { isOpen, onOpen, onClose } = useDisclosure();
 	const cancelRef = React.useRef();
 	const cancelBtnText = props.cancelBtnText || __('No');
+	const cancelButton = <Button buttonText={cancelBtnText} ref={cancelRef} onClick={onClose} />;
 	const header = props.header || __('Are you sure you want to close this?');
 	const okBtnText = props.okBtnText || __('Yes');
+	const okButton = <Button buttonText={okBtnText} buttonType={ButtonType.ACCENT} onClick={props.onClose} ml={3} />;
```
Comment (tn3rb) [inline assets/src/application/ui/layout/modal/ModalWithAlert.tsx:25]: what about using the `useConfirmationDialog` hook that you just added ?

Comment (nicolad) [reply assets/src/application/ui/layout/modal/ModalWithAlert.tsx:25]: will do

Comment (nicolad) [reply assets/src/application/ui/layout/modal/ModalWithAlert.tsx:25]: double tested after change request:


Peek 2020-05-06 22-52

Comment (nicolad) [reply assets/src/application/ui/layout/modal/ModalWithAlert.tsx]: the update: https://github.com/eventespresso/event-espresso-core/pull/2777/commits/3f4a54f4406e86aaca97a47ac44423ab0f8c2268

**Pull request merged**